.circleci/config.yml

```diff
@@ -164,6 +168,37 @@ jobs:
             - ~/.pyenv/versions/
           key: docs-{{ .Branch }}-{{ checksum "setup.cfg" }}-{{ checksum ".readthedocs.yaml" }}
 
+  test-python3-11:
+    description: "Test Python-3.11"
+    executor:
+      name: docker-executor
+      python_version: "3.11"
+    parallelism: 4
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - deps-{{ .Branch }}-{{ checksum "setup.cfg" }}-{{ checksum "/home/circleci/.pyenv/version" }}
+            - deps-main-{{ checksum "setup.cfg" }}-{{ checksum "/home/circleci/.pyenv/version" }}
+      - run:
+          name: Install Dependencies
+          command: pip install -U -e .[test_python_3_11,tests]
+      - run:
+          name: Run tests
+          command: |
+            set -e
+            TEST_FILES=$(circleci tests \
+                          glob "tests/**/test_*.py" | sed '/tests\/apache\/hive/d' \
+                          | circleci tests split --split-by=timings \
+                        )
+            pytest --junit-xml=test-report/report.xml  $TEST_FILES
```

Comment:
I have added on at the top of this job PTAL

Reply:
Aha, I missed the codecov step. I did not include it because except for hive all tests are running here and including it will not change coverage. But happy to include it if you think it makes sense